### PR TITLE
Update the smart contracts overview

### LINKED ISSUE #178: Update obol managers page
# 🎯 Problem to be solved

Our `obol managers` section of docs (https://docs.obol.tech/docs/sc/introducing-obol-managers) is outdated. 

The current `obol-managers` documentation doesn't especially match the actual implementation of the waterfall+splitter factory. We need to update the documentation to reflect the current implementation and verify the source code on etherscan before sharing the addresses.

# 🛠️ Proposed solution

Update this page with the latest developments and architecture details.

We can aim to take inspiration from [this](https://dev.docs.0xsplits.xyz/core).

Regarding waterfall+splitter factory, we need to create new documentation that explains how it works and how to use it. The documentation should include the following sections:
- Introduction: a brief explanation of what the waterfall+splitter factory is and what it does.
- Usage: a detailed explanation of using the waterfall+splitter factory, including code examples and step-by-step instructions.
- Implementation: a technical description of how it is implemented, including the smart contract code and any relevant details.

# 🧪 Tests

None.

# 👐 Additional acceptance criteria

# ❌ Out of Scope

### PR DESCRIPTION
**Pull request by OisinKyne**
## Summary

Updates the smart contract page

ticket: Closes #178

### CODE REVIEW

File: docs/sc/01_introducing-obol-managers.md [hunk 1]
```diff
@@ -31,29 +42,48 @@ contract WithdrawalRecipientOwnable is Ownable {
...
+An Ownable Withdrawal Recipient is the most basic example of a withdrawal recipient contract. It implements Open Zeppelin's `Ownable` interface and allows one address to call the `withdraw()` function, which pulls all ether from the address into the owner's address (or another address specified). This contract does no accounting on the amount of ether that is withdrawn, nor does it differentiate reward from principal.
 
-### Immutable Withdrawal Recipient
+### Optimistic Withdrawal Recipient
 
-An immutable withdrawal recipient is similar to an ownable recipient except the owner is hardcoded during construction and the ability to change ownership is removed. This contract should only be used as part of a larger smart contract system, for example a yearn vault strategy might use an immutable recipient contract as its vault address should never change. 
+This is the primary withdrawal recipient Obol uses, as it allows for the separation of reward from principal, as well as permitting the ongoing withdrawal of accruing rewards.
```
Comment (samparsky) [inline docs/sc/01_introducing-obol-managers.md:49]: should it be "primary" or "consensus layer"

File: docs/sc/01_introducing-obol-managers.md [hunk 2]
```diff
@@ -31,29 +42,48 @@ contract WithdrawalRecipientOwnable is Ownable {
...
+Worst-case mass slashings can theoretically exceed 16 ether, if this were to occur, the returned principal would be misclassified as a reward, and distributed to the wrong address. This risk is the drawback that makes this contract variant 'optimistic'. If you intend to use this contract type, **it is important you understand and accept this risk**, however minute.
 
-### Validator Registry
+The alternative is to use an 0xSplits waterfall contract, which won't allow the claiming of rewards until all principal ether has been pulled, meaning validators need to be exited for operators to claim their CL rewards.
 
-If validators need to be managed on chain programmatically rather than manually with humans triggering exits, a validator registry can be used. Deposits getting activated get an entry into the validator registry, and validators using 0x03 exits get staged for removal from the registry. This registry can be used to coordinate many validators with similar operators and configuration. 
+:::
```
Comment (samparsky) [inline docs/sc/01_introducing-obol-managers.md:65]: should we include the math for the worst case scenario here?

File: docs/sc/01_introducing-obol-managers.md [hunk 3]
```diff
@@ -31,29 +42,48 @@ contract WithdrawalRecipientOwnable is Ownable {
...
 
-### Deposit Registry
+- A _principal_ address: The address that controls where the principal ether will be transferred post-exit.
+- A _reward_ address: The address where the accruing reward ether is transferred to.
+- The amount of ether that makes up the principal.
 
-The Deposit Registry is a way for the deposit and activation of distributed validators to be two separate processes. In the simple case for DVs, a registry of deposits is not required. However when the person depositing the ether is not the same entity as the operators producing the deposits, a coordination mechanism is needed to make sure only one 32 eth deposit is submitted per DV. A deposit registry can prevent double deposits by ordering the allocation of ether to validator deposits. 
+This contract **assumes that any ether that has appeared in it's address since it was last able to do balance accounting is reward from a successful validator** (or number of validators) unless the change is > 16 ether. This means balance skimming is immediately claimable as reward, while an inflow of e.g. 31 ether is tracked as a return of principal (despite being slashed in this example).
```
Comment (samparsky) [inline docs/sc/01_introducing-obol-managers.md:57]: i believe including a diagram here could help

File: docs/sc/01_introducing-obol-managers.md (hunk 1 above)
Comment (OisinKyne) [reply docs/sc/01_introducing-obol-managers.md:49]: I meant it to mean that we don't actually use the Ownable Withdrawal Recipient at all, and that it is only there as a primer to give people an idea how the more complex ones work. We could arguably remove the ownable one at some point, but i thought it was clean so at the time left it in. Is that a bad idea?

File: docs/sc/01_introducing-obol-managers.md (hunk 3 above)
Comment (OisinKyne) [reply docs/sc/01_introducing-obol-managers.md:57]: I'd be interested to see how we could visualise this, its definitely not super well explained. Do you want to do a rough sketch to show me what you are thinking and then we can make a pretty variant second?

File: docs/sc/01_introducing-obol-managers.md (hunk 2 above)
Comment (OisinKyne) [reply docs/sc/01_introducing-obol-managers.md:65]: Yeah, do you think you could work out what percentage of the network needs to be slashed for the penalty to be > 16 ether? Its not the easiest thing to calculate, but it definitely would help people understand how big an issue would have to happen to expose them to this risk.

Comment (samparsky) [reply docs/sc/01_introducing-obol-managers.md:65]: yes, I will share it

File: docs/sc/01_introducing-obol-managers.md (hunk 3 above)
Comment (samparsky) [reply docs/sc/01_introducing-obol-managers.md:57]: ok, I will work on it

**Comment by OisinKyne** [comment]
Merging now so that I can show it to a client. We can add the diagram and equation in future when you get the time to develop it @samparsky

**Pull request merged**